### PR TITLE
 Make TRUNCATECOLUMNS optional, also add all other parameters

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
this breaks backwards comparability by removing some defaults eg:

```python
        truncate_columns=True,
        delimiter=',',
        ignore_header=0,
        empty_as_null=True,
        blanks_as_null=True,
```

**Comment by graingert** [comment]
@jklukas I was wondering if I could have a set of AWS credentials that can only upload to an S3 bucket?

**Comment by graingert** [comment]
This needs a changelog update, and the docs updating.

File: redshift_sqlalchemy/commands.py [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+import sqlalchemy as sa
+from sqlalchemy.ext import compiler as sa_compiler
+from sqlalchemy.sql import expression as sa_expression
+
+
+# At the time of this implementation, no specification for a session token was
+# found. After looking at a few session tokens they appear to be the same as
+# the aws_secret_access_key pattern, but much longer. An example token can be
+# found here:
+#   http://docs.aws.amazon.com/STS/latest/APIReference/API_GetSessionToken.html
```
Comment (MichaelAquilina) [inline redshift_sqlalchemy/commands.py]: thats pretty bad tom. http is deprecated

File: redshift_sqlalchemy/commands.py [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+# The regexs for access keys can be found here:
+#     http://blogs.aws.amazon.com/security/blog/tag/key+rotation
```
Comment (MichaelAquilina) [inline redshift_sqlalchemy/commands.py]: omg tom

**Comment by MichaelAquilina** [comment]
Looks good to me :+1:

**Comment by graingert** [comment]
@jklukas @thisfred @bouk @cpcloud: Thoughts?

File: tests/test_copy_command.py [hunk 3]
```diff
@@ -98,16 +129,66 @@ def test_invalid_compression():
...
+    expected_result = """
+    COPY schema1.t1 (col1, col2) FROM 's3://mybucket/data/listing.manifest'
+    WITH CREDENTIALS AS '%s'
+    FORMAT AS JSON AS 's3://mybucket/data/jsonpath.json'
+    GZIP
+    MANIFEST
+    ACCEPTANYDATE
+    TIMEFORMAT AS 'auto'
+    """ % creds
+    copy = dialect.CopyCommand(
```
Comment (graingert) [inline tests/test_copy_command.py]: analogous code appears in the YPlan test suite but hits S3 and Redshift

**Comment by graingert** [comment]
@jklukas @thisfred @bouk @cpcloud I'm going to go ahead and merge this and make a release because I need it in production pretty sharpish

**Pull request merged**